README.md

```diff
@@ -203,27 +214,31 @@ $ ./ysoserial.exe --credit
 ysoserial.net has been developed by Alvaro Muñoz (@pwntester)
 
 Credits for available formatters:
-	ActivitySurrogateDisableTypeCheck
+	ActivitySurrogateDisableTypeCheck [derived method]
 		Nick Landers
 	ActivitySurrogateSelector
 		James Forshaw
 	ActivitySurrogateSelectorFromFile
 		James Forshaw
+	DataSet [derived method]
+		James Forshaw, implemented by Soroush Dalili
```

Comment:
We dont have that info for other gadgets. Please remove "implemented by" if someone wants to find out, they can always check git

Reply:
I don't think checking git to find who has spent time to implement a new gadget is encouraging.  Perhaps Credit is not a right place to keep this but we will certainly need a place to give credit to those who code new gadgets/plugins although not being their original discovery. 

Similar to metasploit modules, we will need an `Author` in addition to `Credit`. I would personally like to add `References` to each gadget and plugin as well to give a reference to an original research or the CVE.

What shall we do here do you think?

Reply:
If you think this is important, we can add who the implementor is (I think author may be confusing) to the gadget code in the form of a comment. But my main point is to try to keep main command output cleaner. I dont think that is the best place to credit all contributions.

Reply:
As you have seen it, it does not show the names in the command line anymore unless you state it using --credit argument. I think having references and credit is good. Have you found metasploit modules confusing in terms of who has written them and what's being exploited and who has found the issue originally? here is an example: https://www.rapid7.com/db/modules/exploit/windows/browser/adobe_toolbutton (I did not write the module in that case but I was the finder) - perhaps if we add all the names to the credit section then that will make it easier (without saying who has done what?)?

Reply:
that is a good example of why authors is a bit confusing. I dont know if it was you or Juan the one that discovered the issue. But Im ok using author as long as we use something else for other people contributing to it. What about "Authors" and "Contributors"?

Reply:
I would go with "Finders" and "Contributors" in the credit section - like 
```
Credit: [Finders: John Doe] [Contributors: James Doe] 
```
Or if you don't like freestyling, we can make them part of the gadget objects (separately)

Reply:
Thanks ok, thanks!